File: projeuler/go/p002.go
package main

import "fmt"

func main() {
	i := 1
	j := 2
	t := 0
	n := 0
	for {
		if (j % 2) == 0 {
			n += j
		}
		t = j
		j += i
		i = t
		if j > 4000000 {
			break
		}
	}
	fmt.Printf("Problem 2: %d\n", n)
}

File: projeuler/c/p002.c
#include <stdio.h>

int main()
{
	int prev = 1;
	int cur = 2;
	int sum = 0;

	while (cur < 4000000) {
		sum += cur;

		prev += 2 * cur;
		cur = 2 * prev - cur;
	}

	printf("Problem 2: %d\n", sum);
	return 0;
}